perllib/FixMyStreet/Cobrand/Surrey.pm

```diff
@@ -200,18 +200,18 @@ sub dashboard_export_problems_add_columns {
 
     $csv->add_csv_columns(
         alerts_count => "Subscribers",
+        external_id  => "External ID",
     );
 
     my $alerts_lookup = $csv->dbi ? undef : $self->csv_update_alerts;
 
     $csv->csv_extra_data(sub {
         my $report = shift;
 
-        if ($alerts_lookup) {
-            return { alerts_count => ($alerts_lookup->{$report->id} || 0) };
-        } else {
-            return { alerts_count => ($report->{alerts_count} || 0) };
-        }
+        return {
+            alerts_count => ($alerts_lookup ? $alerts_lookup->{$report->id} || 0 : $report->{alerts_count} || 0),
+            external_id  => $csv->dbi ? $report->{external_id} : $report->external_id,
```

Comment:
Is this right? I notice that Northamptonshire just does `return if $csv->dbi;`, but couldn't work our why 😕

Reply:
The Script::CSVExport gets all the columns on a report by default in its hashref, but the non-pre-gen only gets a subset. So you don't need to look up anything that's a column of the DB when $csv->dbi is set, you could if $alerts_lookup is set add external_id there, and if not set return only alerts_count as it does currently.